pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1117,16 +1118,16 @@ func (s *AWSCloud) getSelfAWSInstance() (*awsInstance, error) {
 		if err != nil {
 			return nil, fmt.Errorf("error fetching instance-id from ec2 metadata service: %v", err)
 		}
-		privateDnsName, err := s.metadata.GetMetadata("local-hostname")
+		instance, err := s.getInstanceByID(instanceId)
```

Comment:
Maybe leave the above line (but commented out), and explain why we have to do this in a comment?